pkg/speccheck/speccheck.go

```diff
@@ -102,9 +92,73 @@ func (sc *SpecCheck) compareActualVsSpec(spec, actual *unstructured.Unstructured
 	return actual, true, nil
 }
 
+// removes:
+// - empty slice/map fields
+// - typed nil slice/map fields
+// - nil fields
+func trimEmptyField(u *unstructured.Unstructured, fields ...string) {
+	val, found, err := unstructured.NestedFieldNoCopy(u.Object, fields...)
+	if err != nil || !found {
+		return
+	}
+	remove := false
+	switch typedVal := val.(type) {
+	case map[string]interface{}:
+		if len(typedVal) == 0 { // typed nil/empty map
+			remove = true
+		}
+	case []interface{}:
+		if len(typedVal) == 0 { // typed nil/empty slice
+			remove = true
+		}
+	case nil: // untyped nil
+		remove = true
+	}
+	if remove {
+		unstructured.RemoveNestedField(u.Object, fields...)
+	}
+}
+
+// TODO Is this ok? Check that there is only one controller and it is THIS bundle
+func processOwnerReferences(spec, actual []meta_v1.OwnerReference) []meta_v1.OwnerReference {
+	if len(spec) == 0 {
+		// return nil slice to make it go away
+		return nil
+	}
+	return spec
```

Comment:
Why do we pass `actual` if we never use it?
Is this function properly implemented?

Reply:
So we need to keep non-controller ownerReferences?

Reply:
I've removed unused arguments.